src/platform/forms-system/src/js/review/StringField.jsx

```diff
@@ -13,18 +14,69 @@ import {
  * in the config
  */
 export default function StringField(props) {
-  const { registry, schema, uiSchema, formData } = props;
-  const uiOptions = getUiOptions(uiSchema);
+  const {
+    registry = getDefaultRegistry(),
+    schema,
+    uiSchema,
+    formData,
+    children,
+    formContext,
+    id,
+  } = props;
+  const realUiSchema = checkForNestedUISchema(uiSchema);
+  const uiOptions = getUiOptions(realUiSchema);
   const labels = uiOptions.labels || {};
   const enumOptions = Array.isArray(schema.enum) && optionsList(schema);
+  const isRoot = id === 'root';
+  let Widget = _.get('ui:reviewWidget', realUiSchema);
 
-  let Widget = _.get('ui:reviewWidget', uiSchema);
   if (!Widget) {
-    const defaultWidget = schema.format || (enumOptions ? 'select' : 'text');
-    Widget = getWidget(
-      schema,
-      uiOptions.widget || defaultWidget,
-      registry.widgets,
+    if (schema.type !== 'object' && schema.type !== 'array') {
+      const defaultWidget = schema.format || (enumOptions ? 'select' : 'text');
+      Widget = getWidget(schema, defaultWidget, registry.widgets);
+    } else {
+      return children;
+    }
+  }
+
+  if (
+    (isRoot && schema.type === 'object') ||
+    (isRoot && schema.type === 'array')
+  ) {
+    let title = formContext.pageTitle;
+    if (!formContext.hideTitle && typeof title === 'function') {
+      title = title(formData, formContext);
+    }
```

Comment:
I know this is JS so anything is possible, but I don't like having a variable start out as a function and get reassigned to be a string. Please keep the types of the values consistent.

Additionally, if you look at my following comment I think this can be removed entirely. We are doing the same `!formContext.hideTitle` check below when we render, so we don't need to do it here as well.

Reply:
Are we trying to handle cases here where `title` might be a component?

Reply:
Sure!